borgmatic/config/validate.py

```diff
@@ -50,6 +50,14 @@ def apply_logical_validation(config_filename, parsed_configuration):
             )
         )
 
+    consistency_archive_prefix = parsed_configuration.get('consistency', {}).get('archive_prefix')
+    if archive_name_format and not consistency_archive_prefix:
+        raise Validation_error(
+            config_filename, (
+                'If you provide an archive_name_format, you must also specify a consistency archive_prefix.',
+            )
+        )
+
```

Comment:
You could in theory omit this validation for the consistency `archive_prefix` value. The check exists for the retention `prefix` so that you don't actually prune archives that you aren't creating as part of your config (potentially resulting in data loss). But unlike pruning, it wouldn't be a huge deal if you accidentally do some extra consistency checks.

I could also see an argument for leaving it in. Let me know which you'd prefer.

Reply:
Well speaking from experience (I introduced this tool into my stack last night), I only realised I needed to set the retention `prefix` value because of the validation error message. 

I feel it would be valuable to new users to have this error be raised. 

As for existing users, however, I imagine there would be complaints (or issues raised) where they upgrade `borgmatic` and find their exiting configs no longer work. 

As you mentioned, perhaps we could have some better messaging around when and why this was introduced, so that existing users getting the error can change their configuration accordingly. 

Maybe something along the lines of:

```
Since version 1.2.0, if you provide `archive_name_format`, you must also specify `consistency.prefix`.
```

I noticed you use semver, so maybe this is acceptable?

Reply:
Sure, that would work. I honestly expect users to read the release notes when upgrading (maybe that's expecting too much!), so a minor incompatibility like that is probably fine.

There is an alternative, which is just making this a warning rather than an outright error if you want the user to be notified without considering this fatal. There's no precedent for that right now, but it would be pretty easy to just print to stderr, for instance.